mne/preprocessing/_css.py

```diff
@@ -0,0 +1,174 @@
+
+
+import numpy as np
+import mne
+import matplotlib.pyplot as plt
+
+
+def _css(mag_data, grad_data, filter_data, r=6):
+    """Remove the cortical contribution to filter_data (typically
+    magnetometer or EEG data) by means of a temporal projection.
+
+    This procedure removes the common signal subspace between mag_data
+    and grad_data from filter_data using r number of projection vectors.
+    Note that the SNR needs to be relatively high for this to work; it
+    therefore makes most sense using this on averaged data in an evoked
+    object.
+
+    Parameters
+    ----------
+    mag_data : np.ndarray of float, shape (n_mag_sensors,n_times)
+        The magnetometer data. Can use either all magnetometer data or
+        a few selected sensors close to a region you want to suppress.
+    grad_data : np.ndarray of float, shape (n_grad_sensors,n_times)
+        The gradiometer data. Can use either all gradiometer data or
+        a few selected sensors close to a region you want to suppress.
+    filter_data : np.ndarray of float, shape (n_grad_sensors,n_times)
+        The data to be filtered, typically the magnetometer or EEG data.
+    r : int
+        The number of projection vectors.
+
+    Returns
+    -------
+    filtered_data : np.ndarray of int, shape (n_features)
+        The outlier indices.
+    """
+    # Orthonormalize gradiometer and magnetometer data by a QR decomposition
+    Qg, rg = np.linalg.qr(grad_data.T)
+    Qm, rm = np.linalg.qr(mag_data.T)
+
+    # Calculate cross-correlation
+    C = np.dot(Qg.T, Qm)
+
+    # Channel weights for common temporal subspace by SVD of cross-correlation
+    Y, S, Z = np.linalg.svd(C)
```

Comment:
we normally use `scipy.linalg` although I'm not sure why :) maybe @larsoner can tell us

and there is `_safe_svd`: https://github.com/mne-tools/mne-python/blob/3a95f054a46abd989a529cd16f4c6eff468ee2de/mne/fixes.py#L28